api/base/serializers.py

```diff
@@ -29,6 +29,65 @@ def _url_val(val, obj, serializer, **kwargs):
         return val
 
 
+class HyperlinkedIdentityFieldWithMeta(ser.HyperlinkedIdentityField):
+    # Returns a list with a url and an optional hash containing additional meta information
+
+    def __init__(self, view_name=None, **kwargs):
+        kwargs['read_only'] = True
+        kwargs['source'] = '*'
+        self.meta = kwargs.pop('meta', None)
+        super(ser.HyperlinkedIdentityField, self).__init__(view_name, **kwargs)
+
+    def get_url(self, obj, view_name, request, format):
+        """
+        Given an object, return the URL that hyperlinks to the object.
+
+        May raise a `NoReverseMatch` if the `view_name` and `lookup_field`
+        attributes are not configured to correctly match the URL conf.
+        """
+        # Unsaved objects will not yet have a valid URL.
+        if obj.pk is None:
+            return None
+
+        lookup_value = getattr(obj, self.lookup_field)
+
+        if lookup_value is None:
+            return None
+
+        kwargs = {self.lookup_url_kwarg: lookup_value}
+        return self.reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+    def to_representation(self, value):
+        request = self.context.get('request', None)
+        format = self.context.get('format', None)
+
+        assert request is not None, (
+            "`%s` requires the request in the serializer"
+            " context. Add `context={'request': request}` when instantiating "
+            "the serializer." % self.__class__.__name__
+        )
+
+        if format and self.format and self.format != format:
+            format = self.format
+
+        # Return the hyperlink, or error if incorrectly configured.
+        try:
+            if self.meta is not None:
+                meta = {}
+                for key in self.meta:
+                    meta[key] = _rapply(self.meta.values()[0], _url_val, obj=value, serializer=self.parent)
+                self.meta = meta
+            return [self.get_url(value, self.view_name, request, format), self.meta]
```

Comment:
Yes, in the fork I overrode the entire `extract_relationships` method.  It was trying to pull out `id` and `type` and nest under `data`.  We didn't want that.  Instead, my method pulls out the url and optional meta information and nests under `links`.